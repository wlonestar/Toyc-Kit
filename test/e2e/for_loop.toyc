#include io

i64 main() {
  for (i64 i = 9; i >= 0; i--) {
    printi64ln(i);
  }
  return 0;
}
