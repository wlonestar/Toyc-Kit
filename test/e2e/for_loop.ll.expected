; ModuleID = 'test/e2e/for_loop.toyc'
source_filename = "test/e2e/for_loop.toyc"
target triple = "x86_64-pc-linux-gnu"

declare i64 @printi64ln(i64)

define i64 @main() {
  %1 = alloca i64, align 8
  store i64 9, ptr %1, align 4
  %2 = alloca i64, align 8
  store i64 9, ptr %2, align 4
  br label %3

3:                                                ; preds = %6, %0
  %4 = load i64, ptr %2, align 4
  %5 = icmp sge i64 %4, 0
  br i1 %5, label %6, label %11

6:                                                ; preds = %3
  %7 = load i64, ptr %2, align 4
  %8 = call i64 @printi64ln(i64 %7)
  %9 = load i64, ptr %2, align 4
  %10 = sub nsw i64 %9, 1
  store i64 %10, ptr %2, align 4
  br label %3

11:                                               ; preds = %3
  ret i64 0
}
